### PR TITLE
Add Tests for `Configuration`

### PR DESCRIPTION
**Pull request by awoc**
# Add Tests for `Configuration`

## :recycle: Current situation

At the moment many tests for `Configuration`s are missing.

## :bulb: Proposed solution

This PR adds tests for the `ConfigurationBuilder`, `APNSConfiguration`, and `FCMConfiguration`.

### Problem that is solved

Improves code coverage of the project.

### Implications

The method `buildIf` in the `ConfigurationBuilder` was changed to either return the `Configuration` within the branch if the condition evaluates to `true` or an `EmptyConfiguration`. This was changed since an `Optional<Configuration>` can't be used in the `ConfigurationCollection`.

### CODE REVIEW

File: Tests/ApodiniTests/ConfigurationTests/APNSConfigurationTests.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+@testable import Apodini
+import XCTest
+import XCTVapor
+import Foundation
```
Comment (lschlesinger) [inline Tests/ApodiniTests/ConfigurationTests/APNSConfigurationTests.swift:4]: do you need to import `Foundation` here?

File: Tests/ApodiniTests/ConfigurationTests/FCMConfigurationTests.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+@testable import Apodini
+import XCTest
+import XCTVapor
+import Foundation
```
Comment (lschlesinger) [inline Tests/ApodiniTests/ConfigurationTests/FCMConfigurationTests.swift:4]: see above

File: Tests/ApodiniTests/ConfigurationTests/FCMConfigurationTests.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import FCM
+
+class FCMConfigurationTests: ApodiniTests {
+    let currentPath = URL(fileURLWithPath: #file).deletingLastPathComponent().path
+    
+    func testMissingFile() throws {
+        XCTAssertRuntimeFailure(FCMConfiguration("something").configure(self.app), "FCM file doesn't exists at path: something")
```
Comment (lschlesinger) [inline Tests/ApodiniTests/ConfigurationTests/FCMConfigurationTests.swift:11]: typo _exist_

**Review by lschlesinger — APPROVED** [review]
Just found two minor points ;) Besides, I think it's a good improvement to the test coverage 👍

**Comment by awoc** [comment]
Sorry I had auto-merge on :D

File: Tests/ApodiniTests/ConfigurationTests/APNSConfigurationTests.swift (hunk 1 above)
Comment (awoc) [reply Tests/ApodiniTests/ConfigurationTests/APNSConfigurationTests.swift:4]: You're absolutely right 👍

File: Tests/ApodiniTests/ConfigurationTests/FCMConfigurationTests.swift (hunk 3 above)
Comment (awoc) [reply Tests/ApodiniTests/ConfigurationTests/FCMConfigurationTests.swift:11]: Thank you for pointing this out

**Comment by bauer-andreas** [comment]
This PR resulted in a warning printed by the compiler 
```
warning: found 4 file(s) which are unhandled; explicitly declare them as resources or exclude from the target
    ./Apodini/Tests/ApodiniTests/ConfigurationTests/mock.pem
    ./Apodini/Tests/ApodiniTests/ConfigurationTests/mock_invalid_fcm.json
    ./Apodini/Tests/ApodiniTests/ConfigurationTests/mock_fcm.json
    ./Apodini/Tests/ApodiniTests/ConfigurationTests/mock.p8
```

Nothing too dramatic but maybe something we wan't to silence?

**Comment by awoc** [comment]
@Supereg I also just noticed that. I fixed this error and incorporated the other feedback with #90

**Pull request merged**